Providers/Profile.swift

```diff
@@ -443,27 +443,69 @@ open class BrowserProfile: Profile {
         return self.remoteClientsAndTabs.insertOrUpdateTabs(tabs)
     }
 
-    public func sendItems(_ items: [ShareItem], toClients clients: [RemoteClient]) -> Deferred<Maybe<SyncStatus>> {
+    public func sendItem(_ item: ShareItem, toClients clients: [RemoteClient]) -> Success {
+        guard let account = self.getAccount() else {
+            return deferMaybe(NoAccountError())
+        }
+
         let scratchpadPrefs = self.prefs.branch("sync.scratchpad")
         let id = scratchpadPrefs.stringForKey("clientGUID") ?? ""
-        let commands = items.map { item in
-            SyncCommand.displayURIFromShareItem(item, asClient: id)
-        }
-        
-        func notifyClients() {
-            let deviceIDs = clients.compactMap { $0.fxaDeviceId }
-            guard let account = self.getAccount() else {
-                return
+        let command = SyncCommand.displayURIFromShareItem(item, asClient: id)
+        let fxaDeviceIds = clients.compactMap { $0.fxaDeviceId }
+
+        // If FxA Messages (Pushbox) is not enabled for this build, simply send the
+        // tabs using the old mechanism via Sync.
+        guard AppConstants.MOZ_FXA_MESSAGES else {
+            return self.remoteClientsAndTabs.insertCommands([command], forClients: clients) >>> {
+                self.syncManager.syncClients() >>> {
+                    account.notify(deviceIDs: fxaDeviceIds, collectionsChanged: ["clients"], reason: "sendtab")
+                    return succeed()
+                }
             }
-            
-            account.notify(deviceIDs: deviceIDs, collectionsChanged: ["clients"], reason: "sendtab")
         }
-        
-        return self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> {
-            let syncStatus = self.syncManager.syncClients()
-            syncStatus >>> notifyClients
-            return syncStatus
+
+        // TODO: If the RemoteDevice has the FxA Messages "availableCommands",
+        // send the tab via that instead.
+        // (e.g. https://hg.mozilla.org/mozilla-central/rev/8ef95c56df04#l2.16)
+        let deferredRemoteDevices = fxaDeviceIds.map { fxaDeviceId in
+            return self.remoteClientsAndTabs.getRemoteDevice(fxaDeviceId: fxaDeviceId)
```

Comment:
This is hitting the database a number of times. Suggest writing a more suitable query.